src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -34,6 +34,15 @@ public async Task Invoke(HttpContext httpContext)
                 return;
             }
 
+            // don't do anything extra if downstream route is single
+            if (httpContext.Items.DownstreamRouteHolder().Route.DownstreamRoute.Count == 1)
```

Comment:
because you use a lot of times "httpContext.Items.DownstreamRouteHolder()." in method, you could, for a little improvement set it in var
```c#
var downstreamRouteHolder=httpContext.Items.DownstreamRouteHolder();
```